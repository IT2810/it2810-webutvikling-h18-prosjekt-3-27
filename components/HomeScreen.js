import React, { PureComponent } from 'react';
import {StyleSheet, Text, View} from 'react-native';
import Icon from "react-native-vector-icons/Ionicons";
import Link from "./StyledText/Link";

export default class HomeScreen extends PureComponent {

  static navigationOptions = {
    title: "Home",
    headerTitleStyle: {
      textAlign: "center",
      flex: 1
    }
  };

  render() {
    return (

      <View style={styles.container}>

        <View style={styles.iconStyle}>
          <Icon name="ios-home" color="green" size={100}/>
        </View>
        <View style={styles.textContainer}>
          <Text style={styles.welcome}>
            Welcome to The Motivation App
          </Text>
          <Text style={styles.textStyle}>
            Tight schedule? Having a hard time keeping track of whats when?
            Organize your schedule with our <Link onPress={this.gotoCalendar}>calendar!</Link>
          </Text>
          <Text style={styles.textStyle}>
            Problems organizing your tasks?
            Try our <Link onPress={this.gotoTodo}>todo list!</Link>
          </Text>
          <Text style={styles.textStyle}>
            Always forgetting your friend's last name? And what was her number again...?
            Use our <Link onPress={this.gotoContacts}>contact list</Link> to keep track of who's who.
          </Text>
          <Text style={styles.textStyle}>
            Tired of counting your steps? Getting distracted and losing count?
            Use our <Link onPress={this.gotoPedometer}>pedometer integration</Link> to
            handle the numbers for you!
          </Text>
        </View>

      </View>
    );
  }

  gotoCalendar = () => {
    this.props.navigation.navigate("Calendar");
  };

  gotoTodo = () => {
    this.props.navigation.navigate("Todo");
  };

  gotoContacts = () => {
    this.props.navigation.navigate("Contacts");
  };

  gotoPedometer = () => {
    this.props.navigation.navigate("Pedometer");
  };
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white",
  },
  textContainer: {
    alignItems: "center",
    marginLeft: 20,
    marginRight: 20,
  },
  welcome: {
    fontSize: 25,
    color: "green",
    textAlign: 'center',
    margin: 10,
  },
  textStyle: {
    fontSize: 15,
    margin: 10,
    color: "grey",
    textAlign: 'center',
  }
});
