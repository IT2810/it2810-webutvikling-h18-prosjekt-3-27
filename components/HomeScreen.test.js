import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import HomeScreen from "./HomeScreen";

it("renders correctly", () => {
  const renderer = new ShallowRenderer();
  const tree = renderer
    .render(<HomeScreen/>);
  expect(tree).toMatchSnapshot();
});
