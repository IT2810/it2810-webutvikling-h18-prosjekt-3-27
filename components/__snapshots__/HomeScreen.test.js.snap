// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component>
    <Icon
      allowFontScaling={false}
      color="green"
      name="ios-home"
      size={100}
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "marginLeft": 20,
        "marginRight": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "fontSize": 25,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Welcome to The Motivation App
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Tight schedule? Having a hard time keeping track of whats when? Organize your schedule with our 
      <Link
        onPress={[Function]}
      >
        calendar!
      </Link>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Problems organizing your tasks? Try our 
      <Link
        onPress={[Function]}
      >
        todo list!
      </Link>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Always forgetting your friend's last name? And what was her number again...? Use our 
      <Link
        onPress={[Function]}
      >
        contact list
      </Link>
       to keep track of who's who.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Tired of counting your steps? Getting distracted and losing count? Use our 
      <Link
        onPress={[Function]}
      >
        pedometer integration
      </Link>
       to handle the numbers for you!
    </Text>
  </Component>
</Component>
`;
